cmd/kube-controller-manager/app/options/attachdetachcontroller.go

```diff
@@ -37,6 +37,7 @@ func (o *AttachDetachControllerOptions) AddFlags(fs *pflag.FlagSet) {
 
 	fs.BoolVar(&o.DisableAttachDetachReconcilerSync, "disable-attach-detach-reconcile-sync", false, "Disable volume attach detach reconciler sync. Disabling this may cause volumes to be mismatched with pods. Use wisely.")
 	fs.DurationVar(&o.ReconcilerSyncLoopPeriod.Duration, "attach-detach-reconcile-sync-period", o.ReconcilerSyncLoopPeriod.Duration, "The reconciler sync wait time between volume attach detach. This duration must be larger than one second, and increasing this value from the default may allow for volumes to be mismatched with pods.")
+	fs.BoolVar(&o.DisableForceDetachOnTimeout, "disable-force-detach-on-timeout", false, "Prevent force detaching volumes based on maximum unmount time and node status. If this flag is set to true, the non-graceful node shutdown feature must be used to recover from node failure (see https://t.ly/Gl1qo). If this flag is set to true and a pod must be forcibly terminated at the risk of corruption, then the appropriate VolumeAttachment object (see here: https://t.ly/-atgq) must be deleted.")
```

Comment:
What's the `https://t.ly` ? It's not a kubernetes official link. And I don't think it's necessary to add this link in the description because the non-graceful node shutdown feature is enough to users. 


> If this flag is set to true and a pod must be forcibly terminated at the risk of corruption, then the appropriate VolumeAttachment object (see here: https://t.ly/-atgq) must be deleted."

It seems to explain the details of the non-graceful node shutdown feature. I think it's not necessary to add because it doesn't relate to the flag itself. If users wants to know the details, they can search it in the kubernetes official document.

How about it?

```
Prevent force detaching volumes when maximum unmount timeout is reached. If true, the non-graceful node shutdown feature should be used to recover from node failure.
```

Reply:
Thanks for the feedback.

To provide some background: I got here because while I was testing this change in our internal repo, I got a review comment from @jingxu97 to include a link to the non-graceful shutdown node feature to help users understand the proper workflow to address hung unmounts if this feature is enabled:

```
If enabled, the non-graceful node shutdown feature could be used to recover from node failure? And add the feature blog link?
Besides non-graceful node shutdown, there is other manual way to just detach volume in case non-graceful node shutdown does not work in certain scenarios.
```

So I added in the two k8s links to resolve those comments:
* Information on non-graceful node shutdown: https://kubernetes.io/blog/2023/08/16/kubernetes-1-28-non-graceful-node-shutdown-ga/
* Information on the CR to remove for manual remediation: https://kubernetes.io/docs/reference/kubernetes-api/config-and-storage-resources/volume-attachment-v1/

I then received feedback from @liggitt indicating that the URLs were making things too long:

```
nit: two urls in the flag help text is probably too long... is there a single short URL we can point to here?
```

I was hoping that if I used a URL shortener like http://t.ly, I'd still be able to keep the two links that I had been earlier asked to include.

And so I replaced the two links with shorter versions:
* https://kubernetes.io/blog/2023/08/16/kubernetes-1-28-non-graceful-node-shutdown-ga/ -> https://t.ly/Gl1qo
* https://kubernetes.io/docs/reference/kubernetes-api/config-and-storage-resources/volume-attachment-v1/ -> https://t.ly/-atgq

@carlory, @jingxu97, @liggitt: Do you folks have any suggestions on how to best reconcile this feedback?

I'm relatively amenable to whatever we decide, but I would like to ensure everyone's concerns are addressed.

Reply:
One question that comes to mind: Is there a preferred URL shortener for the kubernetes project that I should be using in lieu of https://t.ly?

Reply:
https://github.com/kubernetes/website/blob/main/static/_redirects allows creating redirects for the k8s.io domain, so `http://k8s.io/nodes#non-graceful-node-shutdown` could redirect to https://k8s.io/nodes/#non-graceful-node-shutdown

I'd suggest having a single URL in the flag help and making sure the content of that page linked out to relevant APIs or pages.

Reply:
Thank for the clarification @liggitt.

I spent some time poking around the website code to find where to put this.

I think it makes sense to create a small entry under https://github.com/kubernetes/website/tree/main/content/en/docs/tasks, maybe underneath a new `storage` subfolder, called `disable-force-detach-on-timeout` where we point to https://kubernetes.io/blog/2023/08/16/kubernetes-1-28-non-graceful-node-shutdown-ga/ (the preferred remediation if unmount is stuck due to this flag) and https://kubernetes.io/docs/reference/kubernetes-api/config-and-storage-resources/volume-attachment-v1/ (the CR that has to be deleted if the preferred remediation doesn't work or isn't wiling to taint the entire node).

Does this sound like the right way to proceed? If so, I'm hoping I can create a new commit for https://github.com/kubernetes/website with:
* A new page with a very short description of what this flag is and links to the other pages
* A shortlink to the new page to include into the description of this commit

I'll then attach the new commit to this PR assuming I can have commits from different repositories attached to the same PR. If that doesn't work, I'll try creating a new PR.

Please feel free to let me know if I have misunderstood something or if you disagree with this proposed course of action.

Thanks!

Reply:
Apologies for the late response; I was pulled into another issue.

Thanks to some great suggestions from @bswartz, I was able to put together a documentation change here: https://github.com/kubernetes/website/pull/44931

The short redirect link I've picked is `/docs/node-disable-force-detach-on-timeout/` which maps to a short section I've added to `/docs/concepts/architecture/nodes/#disable-force-detach-on-timeout`.

The proposed changes and redirect can currently be tested here:
https://deploy-preview-44931--kubernetes-io-main-staging.netlify.app/docs/node-disable-force-detach-on-timeout/; this currently redirects to https://deploy-preview-44931--kubernetes-io-main-staging.netlify.app/docs/concepts/architecture/nodes/#disable-force-detach-on-timeout.

Reply:
Oh, and the documentation PR can be found here: https://github.com/kubernetes/website/pull/44931

Reply:
I spoke to @bswartz; I think we should be good to go with this change?

@carlory, @liggitt, @msau42: Do the proposed URLs make sense?

In this case, it would be: https://k8s.io/docs/node-disable-force-detach-on-timeout/

If so, does it make sense to commit this before the other documentation change goes in (since I cannot commit both together)? How should we stage this?

Reply:
Note: Based on review feedback, the new link will be https://k8s.io/docs/storage-disable-force-detach-on-timeout/

@carlory, @liggitt, @msau42: Please feel free to let me know if this looks okay; you can see a preview of the page here: https://deploy-preview-45119--kubernetes-io-vnext-staging.netlify.app/docs/storage-force-detach-on-timeout

Reply:
Feedback from SIG Docs: can we use a different short URL?

(I / we didn't know folks were planning to mint one)

See https://github.com/kubernetes/website/pull/45119/files#r1539091498